src/wp-includes/blocks/index.php

```diff
@@ -12,6 +12,82 @@
 require BLOCKS_PATH . 'widget-group.php';
 require BLOCKS_PATH . 'require-dynamic-blocks.php';
 
+/**
+ * Registers core block style handles.
+ *
+ * While {@see register_block_style_handle()} is typically used for that, the way it is
+ * implemented is inefficient for core block styles. Registering those style handles here
+ * avoids unnecessary logic and filesystem lookups in the other function.
+ *
+ * @since 6.3.0
+ */
+function register_core_block_style_handles() {
+	if ( ! wp_should_load_separate_core_block_assets() ) {
+		return;
+	}
+
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
+	}
+
+	$includes_url   = includes_url();
+	$includes_path  = ABSPATH . WPINC . '/';
+	$suffix         = wp_scripts_get_suffix();
+	$wp_styles      = wp_styles();
+	$style_fields   = array(
+		'style'       => 'style',
+		'editorStyle' => 'editor',
+	);
+	$transient_name = 'wp_core_block_styles';
+	$files          = get_transient( $transient_name );
+	if ( ! $files ) {
+		$files = glob( __DIR__ . "/**/**{$suffix}.css" );
+		set_transient( $transient_name, $files );
+	}
+
+	foreach ( $core_blocks_meta as $name => $schema ) {
+		/** This filter is documented in wp-includes/blocks.php */
+		$schema = apply_filters( 'block_type_metadata', $schema );
+
+		// Backfill these properties similar to `register_block_type_from_metadata()`.
+		if ( ! isset( $schema['style'] ) ) {
+			$schema['style'] = "wp-block-$name";
+		}
+		if ( ! isset( $schema['editorStyle'] ) ) {
+			$schema['editorStyle'] = "wp-block-{$name}-editor";
+		}
+
+		foreach ( $style_fields as $style_field => $filename ) {
+			$style_handle = $schema[ $style_field ];
+			if ( is_array( $style_handle ) ) {
+				continue;
+			}
+
+			$style_path = "blocks/{$name}/{$filename}{$suffix}.css";
+			$path       = $includes_path . $style_path;
+
+			if ( ! in_array( $path, $files, true ) ) {
+				$wp_styles->add(
+					$style_handle,
+					false
+				);
+			} else {
+				$wp_styles->add( $style_handle, $includes_url . $style_path );
+				$wp_styles->add_data( $style_handle, 'path', $path );
+
+				$rtl_file = str_replace( "{$suffix}.css", "-rtl{$suffix}.css", $path );
+				if ( is_rtl() && in_array( $rtl_file, $files, true ) ) {
+					$wp_styles->add_data( $style_handle, 'rtl', 'replace' );
+					$wp_styles->add_data( $style_handle, 'suffix', $suffix );
+					$wp_styles->add_data( $style_handle, 'path', $rtl_file );
+				}
+			}
```

Comment:
Using if-else is not a great pattern, and we could easily avoid it here. I think you previously had it like that, not sure why it changed. Better for readability to use early returns than nesting.
```suggestion
			if ( ! in_array( $path, $files, true ) ) {
				$wp_styles->add(
					$style_handle,
					false
				);
				continue;
			}

			$wp_styles->add( $style_handle, $includes_url . $style_path );
			$wp_styles->add_data( $style_handle, 'path', $path );

			$rtl_file = str_replace( "{$suffix}.css", "-rtl{$suffix}.css", $path );
			if ( is_rtl() && in_array( $rtl_file, $files, true ) ) {
				$wp_styles->add_data( $style_handle, 'rtl', 'replace' );
				$wp_styles->add_data( $style_handle, 'suffix', $suffix );
				$wp_styles->add_data( $style_handle, 'path', $rtl_file );
			}
```

Reply:
Fixed.